packages/hardhat/contracts/YourContract.sol

```diff
@@ -10,14 +10,22 @@ contract YourContract {
 
   string public purpose = "Building Unstoppable Apps";
 
+  error EmptyPurposeError(uint code, string message);
+
   constructor() {
     // what should we do on deploy?
   }
 
   function setPurpose(string memory newPurpose) public {
-    purpose = newPurpose;
-    console.log(msg.sender,"set purpose to",purpose);
-    emit SetPurpose(msg.sender, purpose);
-  }
+      if(bytes(newPurpose).length == 0){
+          revert EmptyPurposeError({
```

Comment:
This will also break the subgraph per this issue, which will hopefully be resolved soon https://github.com/graphprotocol/graph-node/issues/2577